samples/3_map_state_simple/src/app/services/map-state.service.ts

```diff
@@ -0,0 +1,9 @@
+export class MapStateService {
+  private _points: __esri.Graphic[] = [];
+  get points() {
+    return this._points;
+  }
+  addPoint(point: __esri.Graphic) {
+    this.points.push(point);
+  }
+}
```

Comment:
I'm of two minds on this.  This does leverage the Typings, which is the Angular II+/TypeScript way of developing.  Which wins purity points.  On the other hand, it causes a deviation from the ArcGIS API docs on how to "do things."  I tend to identify with the neophytes and the middlings  more than the 10-xer's, which is why I chose the method I did I suppose.  Here's a thought.  I wanted to do a simple and an less-simple example of state management, and this may provide an opportunity to show both.  Do the simple with "standard" API syntax and the advanced using the TypeScript approach, sort of a progressivist approach.  This would provide the opportunity to make the distinction in the presentation.  I'm sure the those of us in the audience that dwell in the first five sigmas would appreciate making a clear distinction.